### DISCUSSION TITLE
Proxmox-Updater

### DISCUSSION THREAD

**Discussion by BassT23** [tteck/Proxmox#1244] [General]
Hi @tteck,

first I want to say thanks for your inspiration.
I used your update-lxcs.sh as base for my idea to make the Proxmox-Updater (with the help and support from @Uruknara).

For now, your script is rise from 82 to 650 lines (the main script in development).

Would be cool, if you can make a short info or link in your script page to the [Proxmox-Updater](https://github.com/BassT23/Proxmox)

Spezial Thanks!

**Comment by tteck** [comment]
Is the script still blindly updating Proxmox?

**Reply by BassT23** [reply]
what do you mean with "blindly?

It can update the Hosts / LXC / VM (with QEMU or SSH) in an hole Cluster, if available

**Comment by tteck** [comment]
As stated here: https://community.home-assistant.io/t/proxmox-helper-scripts-other-news/398914/37?u=tteck
>You have clearly invested a significant amount of time and effort into this project, and I want to acknowledge and commend you for that. However, I would advise you to avoid attempting to accomplish too much with a single script. Additionally, I strongly recommend against blindly updating Proxmox.
Proxmox’s update panel provides the option to review the changelog prior to upgrading, and I highly recommend that you take advantage of this feature. It’s important to carefully examine any changes before proceeding with the update to ensure that there are no potential issues or conflicts with your current setup.

In my opinion, Proxmox should never be updated unattended/uninformed, except for after a fresh installation.

**Reply by BassT23** [reply]
I respect your incest, but what the script made is "only" security updates on Proxmox Host - and there only the activated.
If you don't set any risky apt sources there should no problems.
So it is also not for "Headless - run at night".
It is more for make updates smarter.
It has an build in exit code, for error handling, and not run in total crash.
Also it is build in an log, for read out all output.

So in my opinion it is nearly safe to use.
Feel free, to test it out ;)

**Reply by Gauvino** [reply]
For my opinion maybe @tteck should tried it out so I can make is own opinion, but for my side I think we should more documenting the read about the script and warn the user.

**Comment by tteck** [comment]
After reading the code, I've formed my opinion, but I would still suggest using the "Proxmox LXC Updater" script provided in this repository. It offers complete functionality without requiring you to download or install any additional programs or configure any files. With just one command, the script automatically excludes all templates and updates all LXCs, and you can exclude additional containers using the command line if needed.
I believe that your script is attempting to do too many tasks at once, which makes it seem more like a personal updater rather than a tool that can be used by a wider audience.

**Comment by tteck** [comment]
While I don't necessarily think a personal updater is a bad idea, I personally do not want to add it to this repository. Providing support for it would be time-consuming.

**Comment by BassT23** [comment]
Ok, thanks for your inspiration ;)

**Comment by tteck** [comment]
https://forum.proxmox.com/threads/unofficial-updater-script-for-proxmox-ve-host-lxc-vm.121733/post-545385
